dask_jobqueue/sge.py

```diff
@@ -0,0 +1,126 @@
+        option.
+    walltime : str
+        Walltime for each worker job.
+    interface : str
+        Network interface like 'eth0' or 'ib0'.
+    death_timeout : float
+        Seconds to wait for a scheduler before closing workers
+    extra : str
+        Additional arguments to pass to `dask-worker`
+    kwargs : dict
+        Additional keyword arguments to pass to `LocalCluster`
+
+    Examples
+    --------
+    >>> from dask_jobqueue import SGECluster
+    >>> cluster = SGECluster(project='...')
+    >>> cluster.start_workers(10)  # this may take a few seconds to launch
+
+    >>> from dask.distributed import Client
+    >>> client = Client(cluster)
+
+    This also works with adaptive clusters.  This automatically launches and
+    kill workers based on load.
+
+    >>> cluster.adapt()
+    """
+    def __init__(self,
+                 name='dask',
+                 queue='default.q',
+                 project=None,
+                 resource_spec='h_vmem=36GB',
+                 threads_per_worker=4,
+                 processes=9,
+                 memory='7GB',
+                 walltime='0:30:0',
+                 interface=None,
+                 death_timeout=60,
+                 extra='',
+                 **kwargs):
+        self._template = """
+#!/bin/bash
+
+#$ -N %(name)s
+#$ -q %(queue)s
+#$ -P %(project)s
+#$ -l %(resource_spec)s
+#$ -l h_rt=%(walltime)s
+#$ -cwd
+#$ -j y
+
+%(base_path)s/dask-worker %(scheduler)s \
+    --nthreads %(threads_per_worker)d \
+    --nprocs %(processes)s \
+    --memory-limit %(memory)s \
+    --name %(name)s-%(n)d \
+    --death-timeout %(death_timeout)s \
+     %(extra)s
+""".lstrip()
+
+        if interface:
+            host = get_ip_interface(interface)
+            extra += ' --interface  %s ' % interface
+        else:
+            host = socket.gethostname()
+
+        project = project or os.environ.get('SGE_ACCOUNT')
+        if not project:
+            raise ValueError("Must specify a project like `project='UCLB1234' "
+                             "or set SGE_ACCOUNT environment variable")
+        self.cluster = LocalCluster(n_workers=0, ip=host, **kwargs)
+        memory = memory.replace(' ', '')
+        self.config = {'name': name,
+                       'queue': queue,
+                       'project': project,
+                       'threads_per_worker': threads_per_worker,
+                       'processes': processes,
+                       'walltime': walltime,
+                       'scheduler': self.scheduler.address,
+                       'resource_spec': resource_spec,
+                       'base_path': dirname,
+                       'memory': memory,
+                       'death_timeout': death_timeout,
+                       'extra': extra}
+        self.jobs = dict()
+        self.n = 0
+        self._adaptive = None
+        self._submitcmd = 'qsub'
+        self._cancelcmd = 'qdel'
```

Comment:
These should maybe be class variables?

Reply:
Yep, agree for submit and cancel commands.